Index: src/economy.cpp
===================================================================
--- src/economy.cpp	(revision 25493)
+++ src/economy.cpp	(working copy)
@@ -1353,6 +1353,30 @@
 }
 
 /**
+ * Tests, whether vehicle's next order leads to current order.
+ * @param v Vehicle
+ * @return True, if true ...
+ */
+static bool VehicleStayConditional(const Vehicle * v)
+{
+	if (v->GetNumOrders() > 1) {
+		const Order * order = v->GetOrder(v->cur_real_order_index);
+		const Order * next_order = order->next;
+
+		/* Next order must be conditional. */
+		if (next_order != NULL && next_order->IsType(OT_CONDITIONAL)) {
+			const VehicleOrderID ord_id = ProcessConditionalOrder(next_order, v);
+			/* We jump on conditional order to the same station we are currently at. */
+			if (ord_id != INVALID_VEH_ORDER_ID && v->GetOrder(ord_id)->index == order->index) {
+				return true;
+			}
+		}
+	}
+
+	return false;
+}
+
+/**
  * Loads/unload the vehicle if possible.
  * @param front the vehicle to be (un)loaded
  */
@@ -1367,7 +1391,7 @@
 	bool use_autorefit = front->current_order.IsRefit() && front->current_order.GetRefitCargo() == CT_AUTO_REFIT;
 	CargoArray consist_capleft;
 	if (_settings_game.order.improved_load &&
-			((front->current_order.GetLoadType() & OLFB_FULL_LOAD) != 0 || use_autorefit)) {
+			((front->current_order.GetLoadType() & OLFB_FULL_LOAD) != 0 || use_autorefit || VehicleStayConditional(front))) {
 		ReserveConsist(st, front,
 				(use_autorefit && front->load_unload_ticks != 0) ? &consist_capleft : NULL,
 				next_station);
@@ -1622,7 +1646,8 @@
 	if (!anything_unloaded) delete payment;
 
 	ClrBit(front->vehicle_flags, VF_STOP_LOADING);
-	if (anything_loaded || anything_unloaded) {
+	const bool condition_stay = VehicleStayConditional(front);
+	if (!condition_stay && (anything_loaded || anything_unloaded)) {
 		if (_settings_game.order.gradual_loading) {
 			/* The time it takes to load one 'slice' of cargo or passengers depends
 			 * on the vehicle type - the values here are those found in TTDPatch */
@@ -1636,7 +1661,7 @@
 				front->current_order_time >= (uint)max(front->current_order.wait_time - front->lateness_counter, 0)) {
 			SetBit(front->vehicle_flags, VF_STOP_LOADING);
 		}
-	} else {
+	} else if (!condition_stay) {
 		bool finished_loading = true;
 		if (front->current_order.GetLoadType() & OLFB_FULL_LOAD) {
 			if (front->current_order.GetLoadType() == OLF_FULL_LOAD_ANY) {
@@ -1659,7 +1684,7 @@
 		}
 		unloading_time = 20;
 
-		SB(front->vehicle_flags, VF_LOADING_FINISHED, 1, finished_loading);
+		if (finished_loading) SetBit(front->vehicle_flags, VF_LOADING_FINISHED);
 	}
 
 	if (front->type == VEH_TRAIN) {
